homeassistant/components/lgthinq/coordinator.py

```diff
@@ -0,0 +1,225 @@
+                self.device_api.device_id
+            )
+        except ThinQAPIException as exc:
+            if exc.code == ThinQAPIErrorCodes.NOT_CONNECTED_DEVICE:
+                self.is_connected = False
+            return {}
+
+        # Full response into the device api.
+        self.device_api.set_status(data)
+        self.is_connected = True
+        return data
+
+    def __str__(self) -> str:
+        """Return a string expression."""
+        return (
+            f"Coordinator:{self.device_name}"
+            f"(type={self.device_api.device_type}, id={self.device_api.device_id})"
+        )
+
+
+async def async_setup_device_coordinator(
+    hass: HomeAssistant, thinq_api: ThinQApi, device: dict[str, Any]
+) -> list[DeviceDataUpdateCoordinator] | None:
+    """Create DeviceDataUpdateCoordinator and device_api per device."""
+    device_id = device.get("deviceId")
+    if not device_id:
+        _LOGGER.error("Failed to setup device: no device id")
+        return None
+
+    device_info = device.get("deviceInfo")
+    if not device_info:
+        _LOGGER.error("Failed to setup device(%s): no device info", device_id)
+        return None
+
+    # Get an appropriate class constructor for the device type.
+    device_type = device_info.get("deviceType")
+    constructor = DEVICE_TYPE_API_MAP.get(device_type)
+    if constructor is None:
+        _LOGGER.error(
+            "Failed to setup device(%s): not supported device. type=%s",
+            device_id,
+            device_type,
+        )
+        return None
+
+    # Get a device profile from the server.
+    try:
+        profile = await thinq_api.async_get_device_profile(device_id)
+    except ThinQAPIException:
+        _LOGGER.warning("Failed to setup device(%s): no profile", device_id)
+        return None
+
+    device_group_id = device_info.get("groupId")
+
+    # Create new device api instance.
+    device_api: ConnectBaseDevice = (
+        constructor(
+            thinq_api=thinq_api,
+            device_id=device_id,
+            device_type=device_type,
+            model_name=device_info.get("modelName"),
+            alias=device_info.get("alias"),
+            group_id=device_group_id,
+            reportable=device_info.get("reportable"),
+            profile=profile,
+        )
+        if device_group_id
+        else constructor(
+            thinq_api=thinq_api,
+            device_id=device_id,
+            device_type=device_type,
+            model_name=device_info.get("modelName"),
+            alias=device_info.get("alias"),
+            reportable=device_info.get("reportable"),
+            profile=profile,
+        )
+    )
+
+    # Create a list of sub-devices from the profile.
+    # Note that some devices may have more than two device profiles.
+    # In this case we should create multiple lg device instance.
+    # e.g. 'WashTower-Single-Unit' = 'WashTower{dryer}' + 'WashTower{washer}'.
```

Comment:
Is this what you meant with the double device thingy?

Reply:
Washtower is a single unit. But it has two devices(dryer, washer)
thinqconnect library send us as one device unit data for two devices at the same time in dictionary.
For user's own sake, we need to split them into 2 devices with ease to use.
If we don't split them, there might be many entities like power1, power2, mode1, mode2....
We think it could be a burden for a user when they control entities.

Reply:
Ideally every physical device is its own device within Home Assistant. But I can also imagine that if you have a lot of appliances that it will be annoying with regards to entities and entity naming.

Reply:
Let me explain how the washtower works.
When we get the washtower's properties, we get one dictionary which has two dictionary inside for dryer, washer (which has a different sub_id and a same device_id)
If one of dryer states changed, we send it with one of the coordinators with the proper sub_id (maybe first coordinator) and one of washer states changed, we send it with the other coordinator.
When it comes to creating a devices' coordinator(s),
if sub_ids exists create a coordinator for each sub_id and if there is no sub_id, create a single coordinator.